Assets/Shopify/UIToolkit/Theme Controllers/ThemeControllerBase.cs

```diff
@@ -13,25 +14,47 @@ public abstract class ThemeControllerBase : MonoBehaviour, IShopCredentials {
         /// The Shop Domain to connect to, in the form of "myshop.myshopify.com"
         /// </summary>
         public string ShopDomain {
-            get { return _shopDomain; }
+            get { 
+                if (_shopDomain == null) {
+                    _shopDomain = EditorPrefs.GetString(ShopDomainEditorPrefsKey);
+                }
+                return _shopDomain; 
+            }
             set {
                 _shopDomain = value;
                 InvalidateClient();
+                EditorPrefs.SetString(ShopDomainEditorPrefsKey, value);
```

Comment:
I think we should only persist these if the credentials are valid, moving this to the editor gives you a real good spot to hook into there for this case